src/changefile/conventionalCommits.ts

```diff
@@ -0,0 +1,43 @@
+import { ChangeType } from '../types/ChangeInfo';
+
+/**
+ * 1. type
+ * 2. scope
+ * 3. breaking
+ * 4. message
+ */
+const COMMIT_RE = /([a-z]+)(?:\(([a-z]+)\))?(!)?: (.+)/;
+
+interface ConventionalCommit {
+  type: string;
+  scope?: string;
+  breaking: boolean;
+  message: string;
+}
+
+interface Change {
+  type: ChangeType;
+  message: string;
+}
+
+export function parseConventionalCommit(commitMessage: string): Change | undefined {
+  const match = commitMessage.match(COMMIT_RE);
+  const data: ConventionalCommit | undefined = match
+    ? { type: match[1], scope: match[2], breaking: !!match[3], message: match[4] }
+    : undefined;
+  return data && map(data);
+}
+
+function map(d: ConventionalCommit): Change | undefined {
+  if (d.breaking) {
+    return { type: 'major', message: d.message };
+  }
+
+  switch (d.type) {
+    case 'fix':
+    case 'chore':
```

Comment:
Why would `chore` map to `patch` here? When I think of `chore` commits, I usually think of things like bumping a dev dependency or other changes that does not affect the published package.

Also, can you add support `docs` and `test` as well?

Reply:
@tido64 I've been doing some hacking locally to get it "just right" and I think what would be best would be to let the user provide a dictionary of allowed types and the change type (major/minor/patch/none) it should map to. That has worked well for me.

Reply:
Yes, that makes sense. Also, I just saw your comment now, asking the exact same question 🤣